tests/test_spline_interpolation.cpp

```diff
@@ -388,27 +472,36 @@ TEST_F(SplineInterpolationTest, cubic_spline_with_end_point_velocity)
     EXPECT_FLOAT_EQ(joint_velocities[i], 0.0);
   }
 
-  std::ofstream outfile("../test_artifacts/cubic_spline_with_end_point_velocity.txt");
-  for (unsigned int i = 0; i < actual_positions.size(); ++i)
-  {
-    outfile << time_vec[i] << "," << actual_positions[i][0] << "," << actual_positions[i][1] << ","
-            << actual_positions[i][2] << "," << actual_positions[i][3] << "," << actual_positions[i][4] << ","
-            << actual_positions[i][5] << "," << actual_velocities[i][0] << "," << actual_velocities[i][1] << ","
-            << actual_velocities[i][2] << "," << actual_velocities[i][3] << "," << actual_velocities[i][4] << ","
-            << actual_velocities[i][5] << "," << expected_positions[i][0] << "," << expected_positions[i][1] << ","
-            << expected_positions[i][2] << "," << expected_positions[i][3] << "," << expected_positions[i][4] << ","
-            << expected_positions[i][5] << "\n";
-  }
+  // Verify that the full trajectory have been executed
+  double spline_travel_time;
+  data_pkg->getData("output_double_register_1", spline_travel_time);
+  ASSERT_NEAR(spline_travel_time, s_time.back(), 1e-5);
+
+  writeTrajectoryToFile("../test_artifacts/cubic_spline_with_end_point_velocity.csv", time_vec, expected_positions,
+                        actual_positions, actual_velocities, actual_acc, speed_scaling_vec, spline_time);
 }
 
-TEST_F(SplineInterpolationTest, quintic_spline_with_end_point_velocity)
+TEST_F(SplineInterpolationTest, quintic_spline_with_end_point_velocity_with_speedscaling)
 {
+  // Set target speed scaling to 100% as one test change this value
```

Comment:
This is the desctructor comment, should probably be something like
```suggestion
  // Set speed scaling to 25% to test interpolation with speed scaling active
```